### PR TITLE
Add table_changes test for replaced tables

### PR DESCRIPTION
**Pull request by mdesmet**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java [hunk 1]
```diff
@@ -780,6 +780,24 @@ public void testRowLevelDeletesWithTableChangesFunction()
         }
     }
 
+    @Test
+    public void testCreateOrReplaceWithTableChangesFunction()
+    {
```
Comment (findinpath) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java:785]: > Note that the exception is not a Trino exception. 

Why wouldn't this be a `TrinoException` ?

Comment (mdesmet) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java:785]: It is a `IllegalArgumentException`

Comment (findinpath) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java:785]: Please add 

```
        if (!SnapshotUtil.isParentAncestorOf(icebergTable, endSnapshotId, startSnapshotId)){
            throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Starting snapshot (exclusive) %s is not a parent ancestor of end snapshot %s".formatted(startSnapshotId, endSnapshotId));
        }
```

after https://github.com/trinodb/trino/blob/875fe1b8a891e3293f0baf48cb4eea7b815b8b3d/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/tablechanges/TableChangesFunction.java#L116-L117

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java [hunk 2]
```diff
@@ -780,6 +780,24 @@ public void testRowLevelDeletesWithTableChangesFunction()
...
+                "test_table_changes_function_",
+                "AS SELECT nationkey, name FROM tpch.tiny.nation WITH NO DATA")) {
+            long initialSnapshot = getMostRecentSnapshotId(table.getName());
+            assertUpdate("CREATE OR REPLACE TABLE " + table.getName() + " AS SELECT nationkey, name FROM nation", 25);
+            long snapshotAfterFirstCreateOrReplace = getMostRecentSnapshotId(table.getName());
+
+            assertQueryFails(
+                    "SELECT nationkey, name, _change_type, _change_version_id, to_iso8601(_change_timestamp), _change_ordinal " +
+                            "FROM TABLE(system.table_changes(CURRENT_SCHEMA, '%s', %s, %s))".formatted(table.getName(), initialSnapshot, snapshotAfterFirstCreateOrReplace),
+                    "Starting snapshot (exclusive) %s is not a parent ancestor of end snapshot %s".formatted(initialSnapshot, snapshotAfterFirstCreateOrReplace));
```
Comment (findinpath) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java:797]:
```suggestion
                    "Starting snapshot \\(exclusive\\) %s is not a parent ancestor of end snapshot %s".formatted(initialSnapshot, snapshotAfterFirstCreateOrReplace));
```

**Comment by mosabua** [comment]
I assume you are still working on this @mdesmet @findinpath @Praveen2112

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/tablechanges/TableChangesFunction.java [hunk 3]
```diff
@@ -115,6 +116,9 @@ public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransact
 
         checkSnapshotExists(icebergTable, startSnapshotId);
         checkSnapshotExists(icebergTable, endSnapshotId);
+        if (!SnapshotUtil.isParentAncestorOf(icebergTable, endSnapshotId, startSnapshotId)) {
```
Comment (findinpath) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/tablechanges/TableChangesFunction.java:119]: Change the commit message.
We're fixing an existing functionality here, not only adding a test.

**Review by findinpath — APPROVED** [review]

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java [hunk 4]
```diff
@@ -780,6 +780,24 @@ public void testRowLevelDeletesWithTableChangesFunction()
...
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_table_changes_function_",
+                "AS SELECT nationkey, name FROM tpch.tiny.nation WITH NO DATA")) {
+            long initialSnapshot = getMostRecentSnapshotId(table.getName());
+            assertUpdate("CREATE OR REPLACE TABLE " + table.getName() + " AS SELECT nationkey, name FROM nation", 25);
+            long snapshotAfterFirstCreateOrReplace = getMostRecentSnapshotId(table.getName());
+
+            assertQueryFails(
+                    "SELECT nationkey, name, _change_type, _change_version_id, to_iso8601(_change_timestamp), _change_ordinal " +
```
Comment (homar) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java:795]: wouldn't `select * ` be enough here? like that it blurs the real thing a little

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java [hunk 5]
```diff
@@ -780,6 +780,24 @@ public void testRowLevelDeletesWithTableChangesFunction()
...
 
+    @Test
+    public void testCreateOrReplaceWithTableChangesFunction()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_table_changes_function_",
+                "AS SELECT nationkey, name FROM tpch.tiny.nation WITH NO DATA")) {
+            long initialSnapshot = getMostRecentSnapshotId(table.getName());
+            assertUpdate("CREATE OR REPLACE TABLE " + table.getName() + " AS SELECT nationkey, name FROM nation", 25);
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java:791]: How about adding additional operations before & after `CREATE OR REPLACE` statement? e.g. 
1. CREATE TABLE AS SELECT with TestTable
2. INSERT
3. CREATE OR REPLACE TABLE
4. INSERT

For verifying the query success of `table_changes` table function with 1→2 & 3→4.

**Review by ebyhr — APPROVED** [review]

**Comment by mdesmet** [comment]
@ebyhr: I processed all remarks, PTAL

**Review by ebyhr — APPROVED** [review]

**Pull request merged**